### PR TITLE
remove evil global state shared by all schedulers

### PR DESCRIPTION
**Pull request by lavalamp**
Running e2e now. (needed for #5336)

### CODE REVIEW

**Comment by bgrant0607** [comment]
cc @davidopp @mikedanese @abhgupta

**Comment by lavalamp** [comment]
After a ridiculous amount of effort, I finished running e2e and it appears to have passed.

File: plugin/pkg/scheduler/factory/plugins.go [hunk 1]
```diff
@@ -46,20 +60,26 @@ type AlgorithmProviderConfig struct {
 	PriorityFunctionKeys util.StringSet
 }
 
-// Registers a fit predicate with the algorithm registry. Returns the name,
-// with which the predicate was registered.
+// RegisterFitPredicate registers a fit predicate with the algorithm
```
Comment (davidopp) [inline plugin/pkg/scheduler/factory/plugins.go]: Random question: Do we have a standard about whether you re-state the function name in the comment? Personally I don't like this style (I'd prefer "Registers a fit predicate..." as it was before) but I guess we don't have a standard practice.

**Comment by davidopp** [comment]
LGTM 

Personally I think factories are pretty evil too :-). I'm not sure whether this PR is a big net improvement. But if you think it is worth the somewhat extra complexity it's OK with me.

File: plugin/pkg/scheduler/factory/plugins.go (hunk 1 above)
Comment (mikedanese) [inline plugin/pkg/scheduler/factory/plugins.go]: Its a godoc convention. From the blog, a godoc "comment is a complete sentence that begins with the name of the element it describes." govet issues a warning if you don't do this I believe.

Comment (lavalamp) [inline plugin/pkg/scheduler/factory/plugins.go]: It's godoc style to repeat the function name verbatim, for whatever reason. :/

**Comment by lavalamp** [comment]
Yeah I don't love this either but these Lister variables have to get made at scheduler creation time and not when the module in initialized.

**Comment by satnam6502** [comment]
@davidopp it would be great to label things with satus-lgtm -- I use that to scan the PRs to work out which ones may be mergeable. Thanks.

**Comment by davidopp** [comment]
Satnam: The reason I didn't label it LGTM was that I was waiting for Daniel to confirm he thought adding this layer of indirection was worthwhile (sorry my comment with the LGTM wasn't clear).

**Comment by lavalamp** [comment]
Thanks for merging, my #5446 builds on top of this one.

**Pull request merged**